### PR TITLE
Introduce `Queue.Executable.getParentExecutable`

### PR DESCRIPTION
**Pull request by jglick**
Addresses a longstanding API gap affecting interactions with Pipeline `node` blocks: while you could go from a `PlaceholderTask` to the `WorkflowJob`, and from the `PlaceholderExecutable` to the `PlaceholderTask`, there was no reliable way to find the `WorkflowRun`. https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/170 is the implementation.

### Proposed changelog entries

* N/A

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/model/Queue.java
Comment (daniel-beck) [inline core/src/main/java/hudson/model/Queue.java:2086]: What motivated this difference?

Comment (jglick) [reply core/src/main/java/hudson/model/Queue.java:2086]: Just general sanity—it is pretty confusing for a method named to look for the parent of some object to return the same object. I did not care to propagate the weird design.

**Review by timja — APPROVED** [review]

File: core/src/main/java/hudson/model/Queue.java
Comment (daniel-beck) [reply core/src/main/java/hudson/model/Queue.java:2086]:
```suggestion
         * @return a <em>distinct</em> executable (never {@code this}, unlike the default of {@link SubTask#getOwnerTask}!); or null if this executable was already at top level
```

**Review by daniel-beck — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
